### ISSUE TITLE
[BUG] - types not included in "exports" for mjs of package.json

### ISSUE THREAD

**Issue by dtfiedler** [warp-contracts/warp#454]
**Describe the bug**
When using `nodenext` and importing `warp-contracts`, `types` are not availabe as they are not included in the `exports` of `package.json`

Error message:
```
Could not find a declaration file for module 'warp-contracts'. '.../node_modules/warp-contracts/lib/cjs/index.js' implicitly has an 'any' type.
  There are types at '.../node_modules/warp-contracts/lib/types/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'warp-contracts' library may need to update its package.json or typings.
```

**To Reproduce**
tsconfig.json:
```json
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "declaration": true,
    "outDir": "./dist",
    "strict": true,
    "esModuleInterop": true,
    "resolveJsonModule": true,
  },
  "include": ["src"],
  "exclude": ["node_modules", "**/node_modules/**"]
}
```
app.ts:
```ts
import { WarpFactory } from 'warp-contracts/mjs';
```

**Expected behavior**
`types` are included in `exports` of `package.json`

**Desktop (please complete the following information):**
 - OS: [e.g. Linux] - Mac OS Ventura (13.4.1)
 - Node.js version - 18.17
 - warp-contracts SDK version - 1.4.17
 - warp-contracts plugins being used (+ versions) - N/A

**Comment by dtfiedler** [comment]
Updating the package.json locally for `warp-contracts` to something like below fixes the issue:
```json
  ...rest of package.json
  "exports": {
    "./web": "./bundles/web.bundle.min.js",
    "./mjs": {
      "import": "./lib/mjs/index.mjs",
      "types": "./lib/types/index.d.ts"
    },
    ".": "./lib/cjs/index.js"
  },
```

**Comment by dtfiedler** [comment]
you could alternatively do to simplify the exports, although this may be a breaking change if `/mjs` is being used
```
"exports": {
  ".": {
    "import": "./lib/mjs/index.js",
    "require": "./lib/cjs/index.js",
    "browser": "./bundles/web.bundle.min.js"
  },
  "./web": "./bundles/web.bundle.min.js"
```

**Comment by ppedziwiatr** [comment]
fixed in 1.4.18
